pkg/pipeline/resource_pipeline_node.go

```diff
@@ -0,0 +1,204 @@
+}
+
+const nodesUrl = "pipelines/api/v1/nodes"
+
+func pipelineNodeResource() *schema.Resource {
+
+	var nodeSchema = map[string]*schema.Schema{
+		"friendly_name": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			Description:  "The name of the node. Should be prefixed with the project key",
+		},
+		"project_id": {
+			Type:         schema.TypeInt,
+			Required:     true,
+			ValidateFunc: validation.IntAtLeast(0),
+			Description:  "Id of the project where the node will live.",
+		},
+		"node_pool_id": {
+			Type:         schema.TypeInt,
+			Required:     true,
+			ValidateFunc: validation.IntAtLeast(0),
+			Description:  "Id of the node pool where the node will live.",
+		},
+		"is_on_demand": {
+			Type:        schema.TypeBool,
+			Required:    true,
+			Description: "Set to true for dynamic node pool. Set to false for static node pool.",
+		},
+		"is_auto_initialized": {
+			Type:        schema.TypeBool,
+			Required:    true,
+			Description: "Determine auto or manual initialization.",
+		},
+		"ip_address": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			Description:  "TSet the architecture. This is currently limited to x86_64.",
+		},
+		"is_swap_enabled": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "Enable/disable the use of swap space to increase the amount of virtual memory available to the node. ",
+		},
+		"token": {
+			Type:      schema.TypeString,
+			Computed:  true,
+			Sensitive: true,
+		},
+	}
+
+	var unpackNode = func(data *schema.ResourceData) (Node, error) {
+		d := &ResourceData{data}
+
+		node := Node{
+			FriendlyName:      d.getString("friendly_name"),
+			ProjectId:         d.getInt("project_id"),
+			NodePoolId:        d.getInt("node_pool_id"),
+			IsOnDemand:        d.getBool("is_on_demand"),
+			IsAutoInitialized: d.getBool("is_auto_initialized"),
+			IPAddress:         d.getString("ip_address"),
+			IsSwapEnabled:     d.getBool("is_swap_enabled"),
+		}
+		node.SystemPropertyBag.Token = d.getString("token")
+		return node, nil
+	}
+
+	var packNode = func(d *schema.ResourceData, node Node) diag.Diagnostics {
+		var errors []error
+		setValue := mkLens(d)
+
+		errors = setValue("project_id", node.ProjectId)
+		errors = append(errors, setValue("friendly_name", node.FriendlyName)...)
+		errors = append(errors, setValue("node_pool_id", node.NodePoolId)...)
+		errors = append(errors, setValue("is_on_demand", node.IsOnDemand)...)
+		errors = append(errors, setValue("is_auto_initialized", node.IsAutoInitialized)...)
+		errors = append(errors, setValue("ip_address", node.IPAddress)...)
+		errors = append(errors, setValue("is_swap_enabled", node.IsSwapEnabled)...)
+		errors = append(errors, setValue("token", node.SystemPropertyBag.Token)...)
+		log.Println("[TRACE] token in object", node.SystemPropertyBag.Token)
```

Comment:
I'm always wary of logging sensitive data like token. May be only show the first few characters and obfuscate the rest?

Reply:
Yes, please don't